main.go

```diff
@@ -389,6 +396,93 @@ func run(c *cli.Context) error {
 	return nil
 }
 
+// GetAnyConfigData get config data from env or files
+func GetAnyConfigData(c *cli.Context) (interface{}, error) {
+	// Make a map we can use:
+	confMap := make(map[string]interface{})
+	var conf interface{}
+	if c.IsSet("config") {
+		if c.IsSet(FlagConfigData) {
+			return nil, fmt.Errorf("cannot set %s if --config flag is set", FlagConfigData)
+		}
+		// Load Environment file overrides into the OS Environment Scope
+		err := godotenv.Load(c.String("config"))
+		if err != nil {
+			return nil, fmt.Errorf("Error loading .env file:%s", err)
+		}
+		// Now get any environment data (as set from above)
+	}
+	// Copy environment to new untyped map:
+	for k, v := range EnvToMap() {
+		confMap[k] = v
+	}
+	for _, cd := range c.StringSlice(FlagConfigData) {
+		// Support flag sytax --flag scope=file.yaml or --flag file.yaml
+		fields := strings.Split(cd, "=")
+		switch len(fields) {
+		case 1:
+			if len(c.StringSlice(FlagConfigData)) > 1 {
+				return nil, fmt.Errorf(
+					"only support a single unscoped %s or multiple scoped entries",
+					FlagConfigData)
+			}
+			var err error
+			// --flag file.yaml
+			conf, err = GetConfigData(fields[0], true)
+			if err != nil {
+				return nil, err
+			}
+			// Only support a single top scoped data item today
+			return conf, nil
+		case 2:
+			// --flag scope=file.yaml
+			scopedConf, err := GetConfigData(fields[1], false)
+			if err != nil {
+				return nil, err
+			}
+			// Update the scoped data:
+			confMap[fields[0]] = scopedConf
+		default:
+			return nil, fmt.Errorf(
+				"%s flag cannot be parsed; data.yaml or scope=data.yaml expected",
+				cd)
+		}
+	}
+	// Cast the typed map to generic interface
+	conf = confMap
+	return conf, nil
+}
+
+// GetConfigData gets data from a file
+func GetConfigData(f string, mergeEnv bool) (interface{}, error) {
+	var conf interface{}
+	// First load any env data from files
+	logDebug.Printf("Loading config file:%s\n", f)
+	b, err := ioutil.ReadFile(f)
+	if err != nil {
+		return nil, fmt.Errorf("error reading file '%s':%s", f, err)
+	}
+	logInfo.Printf("Loaded config data from %s", f)
+	// Load yaml
+	if err := yaml.Unmarshal(b, &conf); err != nil {
+		return nil, err
+	}
+	// Update any values which DON't exist from environment
+	if mergeEnv && reflect.TypeOf(conf).Kind() == reflect.Map {
+		confTyped := conf.(map[interface{}]interface{})
+		for k, v := range EnvToMap() {
+			if _, ok := confTyped[v]; !ok {
```

Comment:
`confTyped[k]` ?

Reply:
An 'Anytype' var in golang - https://tour.golang.org/methods/14 so `confTyped` is a strongly typed var with conf.

Reply:
oh I see - mmm the tests pass - probably got k and v the wrong way round